### PR TITLE
Add 2d supports for AHNet and VNet

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
### Description
The corresponding consistent validations have been done locally, and unittests are all covered.

In order to use changeable kernel/stride/padding sizes for both 2D and 3D layers, types of the variables for the corresponding layers cannot be declared. Otherwise, type errors may happen. For example,
``conv_type: Type[Union[nn.Conv2d, nn.Conv3d]] = Conv[Conv.CONV, spatial_dims]`` should be modified into:
``conv_type = Conv[Conv.CONV, spatial_dims]``. Otherwise, the input for the parameter ``kernel_size`` in ``conv_type`` cannot be ``(3, 3, 1)[:spatial_dims]`` to make it changeable for both 2D and 3D kernel sizes.
At the beginning, I followed the coding norms of ``densenet.py`` and declared types for all layers, thus I'm not sure whether it is receivable to write sth like the following, since these two lines' styles are different to avoid aforementioned errors from **mypy**. Please correct if it is not feasible, thanks!
``conv_type = Conv[Conv.CONV, spatial_dims]``
 ``norm_type: Type[Union[nn.BatchNorm2d, nn.BatchNorm3d]] = Norm[Norm.BATCH, spatial_dims]``
### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by yiheng-wang-nv** [comment]
@Nic-Ma @wyli @ericspod @benjamin-gorman Hi all, could you please help to review the PR, thanks!

File: monai/networks/blocks/fcn.py [hunk 1]
```diff
@@ -107,10 +108,10 @@ class FCN(nn.Module):
 
     Args:
         out_channels: number of output channels. Defaults to 1.
-        upsample_mode: The mode of upsampling manipulations, there are two choices:
+        upsample_mode: The mode of upsampling manipulations, there are three choices:
             1) "transpose", uses transposed convolution layers.
-            2) "interpolate", uses standard interpolate way.
-            Using the second mode cannot guarantee the model's reproducibility. Defaults to "transpose".
+            2) "bilinear", uses standard interpolate way.
```
Comment (wyli) [inline monai/networks/blocks/fcn.py:113]: line 111 says three options but only 2 of them listed here. I think "standard interpolate way" is not very informative...

File: monai/networks/nets/ahnet.py [hunk 2]
```diff
@@ -272,53 +320,67 @@ class AHNet(nn.Module):
     Anisotropic Hybrid Network (AH-Net).
     The code is adapted from lsqshr's original version:
     https://github.com/lsqshr/AH-Net/blob/master/net3d.py
-    In order to use pretrained weights from 2D FCN/MCFCN, please call the copy_from function.
-    In the default settings (same as the author's official code), the input size of the first two dimensions should be divided
+    The model supports 2D or 3D inputs, as for the original 3D version, in order to use pretrained weights
```
Comment (wyli) [inline monai/networks/nets/ahnet.py:323]: the network is designed for anisotropic kernels, technically the 2D case should use kernels such as 3x1. Using 3x3 conv. kernels lose the key feature of the anisotropic design. This will cause confusions. please consider clarifying this in the docstring, or modify the implementation.

Comment (yiheng-wang-nv) [reply monai/networks/nets/ahnet.py:323]: I see, let me use 3*1 instead.

File: monai/networks/blocks/fcn.py (hunk 1 above)
Comment (yiheng-wang-nv) [reply monai/networks/blocks/fcn.py:113]: Thanks, = = a stupid error, I forgot to modify this place, let me correct it.

File: monai/networks/nets/ahnet.py [hunk 3]
```diff
@@ -25,29 +25,37 @@ class Bottleneck3x3x1(nn.Module):
...
-        self.bn2 = norm3d_type(planes)
-        self.conv3 = conv3d_type(planes, planes * 4, kernel_size=1, bias=False)
-        self.bn3 = norm3d_type(planes * 4)
+
+        self.conv1 = conv_type(inplanes, planes, kernel_size=1, bias=False)
+        self.bn1 = norm_type(planes)
+        self.conv2 = conv_type(
+            planes,
+            planes,
+            kernel_size=(3, 3, 1)[3 - spatial_dims :],
```
Comment (wyli) [inline monai/networks/nets/ahnet.py:47]: this could be `(3, 3, 1)[-spatial_dims:]`

Comment (yiheng-wang-nv) [reply monai/networks/nets/ahnet.py]: thanks, I change them.
